### PR TITLE
Add axe to travis tests

### PR DESCRIPTION
**Pull request by hstandaert**
## Description

* Split gulp task `axe` into multiple
	* `axe:input`
	* `axe:layout`
	* `axe:components`
* Execute build before testing with axe
* Add it to the `yarn test` script (which is executed in travis)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the styleguide CHANGELOG accordingly.
- [ ] I have ran gulp axe on the compiled files and found no errors.

### CODE REVIEW

**Comment by hstandaert** [comment]
@delrueba, I've looked into removing the `definition-list` rule, but it hasn't been updated in axe-core yet... So I'll just leave it in there for now?

**Comment by delrueba** [comment]
> @delrueba, I've look into removing the `definition-list` rule, but it hasn't been updated in axe-core yet... So I'll just leave it in there for now?

Yes, maybe someday...
I see there have been more issues and PR's since my last comment in september about this rule.

File: package.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
     "yarn": "^1.12.0"
   },
   "scripts": {
-    "test": "gulp validate",
+    "test": "gulp validate && gulp axe",
```
Comment (delrueba) [inline package.json]: Why not just add it to the validate task?

Comment (delrueba) [inline package.json:67]: Why not add it to the validate task?

Comment (delrueba) [reply package.json:67]: Nevermind, guessing the precommit would take to long?
Or because you need to build first?

:+1:  carry on :smile:

**Review by delrueba — APPROVED** [review]

File: gulpfile.js [hunk 2]
```diff
@@ -769,14 +770,14 @@ gulp.task('axe:components', callback => {
   return axe(Object.assign({}, axeOptions, {
     saveOutputIn: 'components.json',
     urls: [
-      'build/components/preview/!(input*|*layout*).html'
+      'build/components/preview/!(input*|*layout*|preview*|textarea*|teaser--*).html'
```
Comment (delrueba) [inline gulpfile.js:773]: Aren't teasers components?

Comment (hstandaert) [reply gulpfile.js:773]: That is very much correct! It was included in the original gulp task as well and so I tossed it out at first, but... 

A teaser is an `li` element which is collated in the overview by this function:

https://github.com/StadGent/fractal_styleguide_gent-base/blob/39c6e0a7701c21080bf7473e3b0c5f818d93afec/components/31-molecules/teaser/teaser.config.js#L4-L15

However, you can access the teasers individually which leads to a axe violation stating that an `li` should always be contained in a `ul`. Hence why the `teaser--` urls are skipped.

**Pull request merged**